### PR TITLE
[GP Ether] - Fixes broken Ether on XDAI and other non-supported networks

### PR DESCRIPTION
**Pull request by W3stside**
- Removes Xdai Updater changing the native token as was done in V2
- now extends entire Ether class into GpEther that checks chainId in it's `onChain` method to adapt for xDai (can be extended for others)
- changes import paths to use custom `GpEther`

## Testing
- open cowswap.dev.gnosisdev.com and try switching to KOVAN, ROPSTEN etc, it breaks.
- on XDAI it breaks when selecting ETH as the buy token (shouldn't even be called ETH)

### CODE REVIEW

File: src/state/wallet/hooks.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export function useTokenBalancesWithLoadingIndicator(
     'balanceOf',
     [address],
     undefined,
-    100_000
+    100_000_000
```
Comment (alfetopito) [inline src/state/wallet/hooks.ts:73]: Why this change?
Actually, what's this number for?

**Review by alfetopito — COMMENTED** [review]
Works fine on Mainnet and Rinkeby
Doesn't break when I switch to any other network.
When on xDai, it show "wrong network"

screenshot_2021-07-13_12-34-51

File: src/state/wallet/hooks.ts (hunk 1 above)
Comment (W3stside) [reply src/state/wallet/hooks.ts:73]: Shouldn't be here, I need to remove this

**Comment by W3stside** [comment]
> Works fine on Mainnet and Rinkeby
> Doesn't break when I switch to any other network.
> When on xDai, it shows "wrong network"
> 
> screenshot_2021-07-13_12-34-51

weird, i dont get that on xdai
image

**Comment by alfetopito** [comment]
> weird, i dont get that on xdai
> image

WTF. Now xDai is working. Well, kinda.

App loads, but balance never loads.
There's this console log which I'm not sure is related:
screenshot_2021-07-13_15-36-39

There are no network requests happening. Maybe the RPC bridge I have setup is not working?
This is my xDai config:
screenshot_2021-07-13_15-38-38

**Comment by W3stside** [comment]
> App loads, but balance never loads.
> There's this console log which I'm not sure is related:
> screenshot_2021-07-13_15-36-39
> 
> There are no network requests happening. Maybe the RPC bridge I have setup is not working?
> This is my xDai config:
> screenshot_2021-07-13_15-38-38

yeah i'm also having this issue. maybe something with the multicall contract. need to crosscheck against our v2 version on what changed there

**Comment by alfetopito** [comment]
Besides the multicall issue you fixed on the other PR, I know how to reproduce the `wrong network` thing.

1. With metamask connected to Mainnet, connect the wallet on CowSwap
2. Switch the network to xDai

It displays `wrong network`.

Looking at the debug logs:
screenshot_2021-07-14_15-15-27

You can see the proper id of each network I switched to, except 2 things:
1. Switching to mainnet does not produce a log message
2. Switching to xdai logs `chainId 1`

Reloading the page when MM is already connected to xDai works, though.

**Comment by W3stside** [comment]
> Besides the multicall issue you fixed on the other PR, I know how to reproduce the `wrong network` thing.
> 
> 1. With metamask connected to Mainnet, connect the wallet on CowSwap
> 2. Switch the network to xDai
> 
> It displays `wrong network`.
> 
> Looking at the debug logs:
> screenshot_2021-07-14_15-15-27
> 
> You can see the proper id of each network I switched to, except 2 things:
> 
> 1. Switching to mainnet does not produce a log message
> 2. Switching to xdai logs `chainId 1`
> 
> Reloading the page when MM is already connected to xDai works, though.

did you log this? this log doesn't exist in my branch. i did delete the updater, it's not required anymore. maybe caching?

**Review by alfetopito — APPROVED** [review]
No issues today. Maybe it was caching :shrug: 

Probably unrelated, but I don't see wxDai icon on the Swap, but I do see it on the token list dropdown

screenshot_2021-07-15_07-53-38

**Comment by W3stside** [comment]
@ anyone else please review post merge
